src/main/kotlin/blackjack/domain/Dealer.kt

```diff
@@ -1,10 +1,60 @@
 package blackjack.domain
 
+import blackjack.domain.enums.Condition
+import blackjack.domain.enums.MatchResult
+
 class Dealer(
-    val deck: Deck
-) {
+    name: String = "딜러",
+    cards: Cards,
+    condition: Condition = Condition.STAY,
+    val deck: Deck,
+) : Participant(name, cards, condition) {
+
+    init {
+        val score = cards.calculateScore()
+        initCondition(score)
+    }
+
+    private fun initCondition(score: Score) {
+        if (score.compareTo(Score.STANDARD_CARD_SCORE) == -1 || score.compareTo(Score.STANDARD_CARD_SCORE) == 0) {
+            this.condition = Condition.PLAY
+        }
+    }
 
-    fun drawCardsFromDeck(count: Int): Cards {
+    override fun hit(card: Card) {
+        super.hit(card)
+
+        val score = cards.calculateScore()
+        changeCondition(score)
+    }
+
+    private fun changeCondition(score: Score) {
+        if (score.compareTo(Score.STANDARD_CARD_SCORE) == 1) {
+            this.condition = Condition.STAY
+        } else if (score.compareTo(Score.BLACK_JACK_SCORE) == 1) {
+            this.condition = Condition.BUST
+        }
+    }
+
+    fun draw(count: Int): Cards {
         return deck.drawCard(count)
     }
+
+    fun currentCondition(): Condition {
+        return this.condition
+    }
+
+    fun determineResult(otherScore: Score): MatchResult {
+        val score = cards.calculateScore()
+
+        return when {
+            score.value > otherScore.value -> MatchResult.WIN
+            score.value < otherScore.value -> MatchResult.LOSE
+            else -> MatchResult.DRAW
+        }
+    }
```

Comment:
딜러와 플레이어의 승부를 가리는 것이라면 Score가 아닌 Player를 인자로 받아야 할 것 같아요.
추가로 cards가 아닌 player와 협력하는 방식으로 바꿀 수 있을 것 같습니다.
```suggestion
fun determineResult(other: Player): MatchResult {
        val score = calculateScore()
        val otherScore = other.calculateScore()

        return when {
            score > otherScore -> MatchResult.WIN
            score < otherScore -> MatchResult.LOSE
            else -> MatchResult.DRAW
        }
    }
```

Reply:
플레이어를 인자로 받도록 수정하였습니다!